pkg/csi/cinder/controllerserver.go

```diff
@@ -132,8 +132,19 @@ func (cs *controllerServer) ControllerPublishVolume(ctx context.Context, req *cs
 	instanceID := req.GetNodeId()
 	volumeID := req.GetVolumeId()
 
+	// Publish Volume Info
+	pvInfo := map[string]string{}
+
 	_, err = cloud.AttachVolume(instanceID, volumeID)
 	if err != nil {
+		_, err = cloud.GetInstance(instanceID)
+		if err != nil {
+			glog.V(3).Infof("Attach failed because instance %s doesn't exist. Trying local-attach", instanceID)
```

Comment:
I thought about splitting the plugin but I ended up convincing myself that a single driver is better in this case. The environments I'm trying to cover with this are:

* Kubernetes running in OpenStack managed instance 
* Kubernetes running on nodes not managed by OpenStack
* Kubernetes running on both, independent nodes and OpenStack managed ones.

The configuration option would work fine but, it would add some complexity to the deployment. Let's take the third case, for example. If you'd like to use a DaemonSet, you would need to label the nodes properly to be able to identify which nodes are part of an OpenStack cluster and which arent.

As I write this, I'm thinking about some of the advantages that splitting would give us:

* Separation of concerns
* More control on the deployment
* We could have `cinderclient` installed *only* on the containers that will run in standalone mode.

By reading your comment, it sounds that what really worries you is the way I'm detecting whether the node is part of an OpenStack cluster or not. Admittedly, I'm not a huge fan of calling `GetInstance` but it does the job for this patch. That said, I'd love to find a better way to do that.

I agree that we should do a good cleanup on this code and it sounds like you'd like to work on that. Is this correct? I'd be happy to help on that too.

Would you agree this PR is a good enough first step and that we can iterate over this? I was thinking about creating a few issues to track the improvements we could do here so that it'll become easier to share the load. 

Thanks a bunch for the review, I'll address the comments requests.